### PR TITLE
Promoted Helm chart version to 1.0.0

### PR DESCRIPTION
**Pull request by solovyovk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,39 +15,23 @@ Before you begin, make sure you have the required environment:
...
-Keep in mind that your custom label has to be either updated in `values.yaml`
-(`affinityLabel` key, recommended), or passed to `helm` calls using 
-`--set affinityLabel=proofreader.company-domain.com/app`.
-
-Now, the chart can be installed the usual way using all the defaults:
+The Chart can be installed the usual way using all the defaults:
 ```shell
 git clone https://github.com/WebSpellChecker/wproofreader-helm.git
 cd wproofreader-helm
-helm install --create-namespace --namespace wsc wsc-app-5-x-x wproofreader --set affinityLabel=proofreader.company-domain.com/app
```
Comment (jshaptala) [inline README.md:37]: do you think there is no sense to keep this naming/versioning as a default?

File: README.md [hunk 2]
```diff
@@ -157,18 +200,22 @@ The service might fail to start up properly if misconfigured. For troubleshootin
 There are several options for how to gather needed details:
```
Comment (jshaptala) [inline README.md:200]:
```suggestion
There are several ways to gather necessary details:
```

File: README.md [hunk 3]
```diff
@@ -157,18 +200,22 @@ The service might fail to start up properly if misconfigured. For troubleshootin
...
+where `wsc` is the namespace and `wproofreader-app` – the name of your release, 
+and `wproofreader-app-values.yaml` – name of the file the data will be written to.
 
 2. Extract the full Kubernetes manifest(s) as follows:
 ```shell
-helm get manifest --namespace wsc wsc-app-5-25-0 > manifests.yaml
+helm get manifest --namespace wsc wproofreader-app > manifests.yaml
 ```
 
 If, for any reason, you do not have access to `helm`, same can be accomplished using
```
Comment (jshaptala) [inline README.md:213]:
```suggestion
If you do not have access to `helm`, same can be accomplished using 
```

File: README.md [hunk 4]
```diff
@@ -157,18 +200,22 @@ The service might fail to start up properly if misconfigured. For troubleshootin
...
+and `wproofreader-app-values.yaml` – name of the file the data will be written to.
 
 2. Extract the full Kubernetes manifest(s) as follows:
 ```shell
-helm get manifest --namespace wsc wsc-app-5-25-0 > manifests.yaml
+helm get manifest --namespace wsc wproofreader-app > manifests.yaml
 ```
 
 If, for any reason, you do not have access to `helm`, same can be accomplished using 
 `kubectl`. To get manifests for all resources in `wsc` namespace, use:
```
Comment (jshaptala) [inline README.md:214]:
```suggestion
`kubectl`. To get manifests for all resources in the `wsc` namespace, run:
```

File: README.md [hunk 5]
```diff
@@ -157,18 +200,22 @@ The service might fail to start up properly if misconfigured. For troubleshootin
...
 ```shell
 kubectl get all --namespace wsc -o yaml > manifests.yaml
+```
+3. Get the logs of all the wsproofreader-app pods in the `wsc` namespace:
```
Comment (jshaptala) [inline README.md:218]:
```suggestion
3. Retrieve the logs of all `wsproofreader-app` pods in the `wsc` namespace:
```

File: manifests/deployment_http.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: wproofreader-app
+  labels:
+    helm.sh/chart: wproofreader-1.0.0
+    app.kubernetes.io/name: wproofreader
+    app.kubernetes.io/instance: wproofreader-app
+    app.kubernetes.io/version: "5.34.1"
```
Comment (jshaptala) [inline manifests/deployment_http.yaml:9]: please update this to the latest app - 5.34.3

File: manifests/deployment_http.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      volumes:
+        - name: dictionaries-volume
+          persistentVolumeClaim:
+            claimName: wproofreader-app-dict
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: wproofreader.company-domain.com/app
```
Comment (jshaptala) [inline manifests/deployment_http.yaml:79]: Just a though is to replace "company" with "custom" or "domain-name" as not all our clients are "companies"
this is to be updated everytwhere

File: manifests/secrets.yaml [hunk 8]
```diff
@@ -1,23 +1,27 @@
...
     app.kubernetes.io/name: wproofreader
-    app.kubernetes.io/instance: wsc-app-sample
-    app.kubernetes.io/version: "5.25.0"
+    app.kubernetes.io/instance: wproofreader-app
+    app.kubernetes.io/version: "5.34.1"
+    app.kubernetes.io/managed-by: Helm
 type: Opaque
 data:
-  license: "cVdlUnRZMTIz"
+  license: "FYVD0w46QhygxIPZ6Byo"
```
Comment (jshaptala) [inline manifests/secrets.yaml:13]: I've checked this key and it doesn't look like a real key. Was it a fake one generated for the purpose of this Chart?

File: wproofreader/values.yaml [hunk 9]
```diff
@@ -31,6 +31,10 @@ certMountPath: "/certificate"
 
 # `dictionaries` structure provide storage for custom user dictionaries inside the cluster
 dictionaries:
+  # enable or disable persistentance of Custom Dictionaries feature
```
Comment (jshaptala) [inline wproofreader/values.yaml:34]:
```suggestion
  # Enable or disable persistence of custom dictionary feature
```

File: wproofreader/values.yaml [hunk 10]
```diff
@@ -31,6 +31,10 @@ certMountPath: "/certificate"
...
+  enabled: false
+  # name of existing the PersistentVolumeClaim to use for storing dictionaries
+  existingClaim: ""
   # absolute path of the directory on the Kubernetes node, where the files should be kept
```
Comment (jshaptala) [inline wproofreader/values.yaml:38]:
```suggestion
  # Absolute path of the directory on the Kubernetes node where the files should be kept.
```

File: wproofreader/values.yaml [hunk 11]
```diff
@@ -31,6 +31,10 @@ certMountPath: "/certificate"
...
   # leave empty if no volume for custom dictionaries should be created
```
Comment (jshaptala) [inline wproofreader/values.yaml:39]:
```suggestion
  # Leave empty if no volume for custom dictionaries should be created.
```

File: wproofreader/values.yaml [hunk 12]
```diff
@@ -52,22 +56,37 @@ image:
   # overrides the image tag whose default is the chart appVersion
```
Comment (jshaptala) [inline wproofreader/values.yaml:56]:
```suggestion
  # Overrides the image tag whose default is the chart appVersion.
```

File: wproofreader/values.yaml [hunk 13]
```diff
@@ -52,22 +56,37 @@ image:
...
 # if enabled, requires "resources" (see below)
-autoscaling:
-  enabled: false
-  minReplicas: 1
-  maxReplicas: 5
-  targetCPUUtilizationPercentage: 80
-  targetMemoryUtilizationPercentage: 80
-
-# max resources usage allowed for pod
+# Note: Wproofreader is able to scale only based on CPU usage metric. The targetMemoryUtilizationPercentage is not supported.
```
Comment (jshaptala) [inline wproofreader/values.yaml:78]:
```suggestion
# Note: WProofreader Server can be scaled only based on CPU usage metric. The targetMemoryUtilizationPercentage is not supported.
```

Comment (jshaptala) [reply wproofreader/values.yaml:78]: We need to move it to the main readme as well.

File: wproofreader/values.yaml [hunk 14]
```diff
@@ -52,22 +56,37 @@ image:
...
+#  targetCPUUtilizationPercentage: 80
+
+# Max resources usage allowed for pod
 # have to be specified if HorizontalPodAutoscaling is enabled
 # if none are given, uses all available on host
 # (to add your values remove "{}" form the line below, uncomment and edit the ones further down)
+# Ref: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
 resources: {}
 #  limits:
 #    cpu: 100m
```
Comment (jshaptala) [inline wproofreader/values.yaml:92]: not sure what this if for (allocation) just to make sure our clients don't use these numbers
and for the one below

**Review by jshaptala — COMMENTED** [review]
I've checked the wording, please check my comments.
Also, a couple of things better to move as well to readme - like CPU untilization metric for scaling.

Suggested wording for successful deployment:

WProofreader Server has been deployed successfully!

Quick start:
link

Status:
link

Version:
link

File: manifests/deployment_http.yaml (hunk 7 above)
Comment (solovyovk) [reply manifests/deployment_http.yaml:79]: Updated with "domain-name".

File: manifests/deployment_http.yaml (hunk 6 above)
Comment (solovyovk) [reply manifests/deployment_http.yaml:9]: Updated

File: manifests/secrets.yaml (hunk 8 above)
Comment (solovyovk) [reply manifests/secrets.yaml:13]: Sure, it was fake key. 
Replaced with "your-license-key".

File: wproofreader/values.yaml (hunk 9 above)
Comment (solovyovk) [reply wproofreader/values.yaml:34]: enable, not Enable. 
To keep commenting uniform.

File: wproofreader/values.yaml (hunk 14 above)
Comment (solovyovk) [reply wproofreader/values.yaml:92]: Updated with actual values

File: README.md (hunk 1 above)
Comment (solovyovk) [reply README.md:37]: In my opinion, it is better to have it the way it is. 
The Helm release name is the part of naming convention here and the resources that will be created will contain it in their names. 
So having a version there will confuse more when using `helm update` command or `--set image.tag`, for example. 
But depending on deployment strategy it is also possible to have a version there.

File: wproofreader/values.yaml (hunk 13 above)
Comment (solovyovk) [reply wproofreader/values.yaml:78]: Updated README with "Application scaling" part

File: README.md [hunk 15]
```diff
@@ -90,28 +74,87 @@ The defaults for the DockerHub image are `cert.pem`, `key.pem`, and `/certificat
...
+## Use in Production
+
+For production deployments, it is highly recommended to specify resource requests and limits for your Kubernetes pods. This helps ensure that your applications have the necessary resources to run efficiently while preventing them from consuming excessive resources on the cluster, which can impact other applications.
+This can be configured in the `values.yaml` file under the `resources` section.
+
+### Recommended Resource Requests and Limits
+
+Below are the recommended values for resource requests and limits. These values are based on the minimum requirements for running the WProofreader Server in a production environment. You may need to adjust these values based on your specific requirements and usage patterns, especially when deploying AI-models, to ensure optimal performance and resource utilization.
+
+```yaml
```
[the remark this answers is not in the document]
Comment (solovyovk) [reply README.md:133]: Updated.

File: README.md [hunk 16]
```diff
@@ -88,30 +72,101 @@ The defaults for the DockerHub image are `cert.pem`, `key.pem`, and `/certificat
...
+However, please note that provisioning the Persistent Volume (PV) and PVC for storage backends like NFS is outside the scope of this Chart. You will need to provision the PV and PVC separately according to your storage backend's documentation before using the `dictionaries.existingClaim` parameter.
+
+## Use in production
+
+For production deployments, it is highly recommended **to specify resource requests and limits for your Kubernetes pods**. This helps ensure that your applications have the necessary resources to run efficiently while preventing them from consuming excessive resources on the cluster which can impact other applications.
+This can be configured in the `values.yaml` file under the `resources` section.
+
+### Recommended resource requests and limits
+
+Below are the recommended resource requests and limits for deploying WProofreader Server v5.34.x with English (en_US, en_GB, en_CA, and en_AU) and including features such as a style guide, spelling autocorrect, Named-Entity Recognition (NER), and text autocomplete. These values reflect the minimum requirements for running WProofreader Server in a production environment.
```
Comment (jshaptala) [inline README.md:131]:
```suggestion
Below are the recommended resource requests and limits for deploying WProofreader Server v5.34.x with enabled English dialects (en_US, en_GB, en_CA, and en_AU) for spelling & grammar check using the English AI language model for enhanced and more accurate proofreading. It also includes such features as a style guide, spelling autocorrect, named-entity recognition (NER), and text autocomplete suggestions (text prediction). These values represent the minimum requirements for running WProofreader Server in a production environment.
```

File: README.md [hunk 17]
```diff
@@ -88,30 +72,101 @@ The defaults for the DockerHub image are `cert.pem`, `key.pem`, and `/certificat
...
+Depending on your specific needs and usage patterns, especially if deploying AI language models for enhanced proofreading, you may need to adjust these values to ensure optimal performance and resource utilization.
```
Comment (jshaptala) [inline README.md:132]:
```suggestion
**Note:** Depending on your specific needs and usage patterns, especially when deploying AI language models for enhanced proofreading in other languages, you may need to adjust these values to ensure optimal performance and resource utilization. Alternatively, you can choose the bare-minimum configuration without AI language models. In this case, only algorithmic engines will be used to provide basic spelling and grammar checks.
```

**Review by jshaptala — COMMENTED** [review]
Added new changes.

File: README.md (hunk 16 above)
Comment (solovyovk) [reply README.md:131]: Applied

File: README.md (hunk 17 above)
Comment (solovyovk) [reply README.md:132]: Applied

**Review by jshaptala — APPROVED** [review]
Changes are OK.

**Pull request merged**